container/falcon-container-aks-implementation-guide.md

```diff
@@ -0,0 +1,282 @@
+# Implementation Guide for CrowdStrike Falcon-Container Sensor on Azure AKS Kubernetes cluster
+
+This guide works through creation of a new Kubernetes cluster, deployment of the Falcon-Container Sensor, and demonstration of detection capabilities of Falcon Container Workload Protection.
+
+No prior Kubernetes or Falcon knowledge is needed to follow this guide. First sections of this guide focus on creation of ACR Container registry and an AKS cluster, however, these sections may be skipped if you have access to an existing cluster.
+
+Time needed to follow this guide: 45 minutes.
+
+## Pre-requisites
+
+- Refer to the lastest documentation on the [falcon-container](https://falcon.crowdstrike.com/documentation/146/falcon-container-sensor-for-linux)
+- Existing Azure Subscription and Global Adminstrator permissions
+- You will need a workstation to complete the installation steps below
+  * These steps have been tested on Linux and should also work with OSX
+- Install [docker](https://www.docker.com/products/docker-desktop) container runtime
+- Verify docker daemon is running on your workstation
+- API Credentials from Falcon with Sensor Download Permissions
+  * These credentials can be created in the Falcon platform under Support->API Clients and Keys.
+  * For this step and practice of least privilege, you would want to create a dedicated API secret and key.
+- Azure Cli installed locally and authenticated
+
+```
+az login
+```
+- Install kubectl
+```
+az aks install-cli
+```
+
+## Deployment
+
+
+### Step 1: Setup an Azure Container Registry
+
+- Set your ACR registry name and resource group name into variables
+- Note: The ACR_NAME must be a unique name globally as a DNS record is created to reference the image registry
+```
+CLOUD_REGION=westus
+ACR_NAME=<arc_unique_name>
+RG_NAME=rg_cswest
+```
+- Create the resource group for the ACR and Cluster
+```
+az group create --name $RG_NAME --location $CLOUD_REGION
+```
+- Create the Azure Container Registry
+```
+az acr create --name $ACR_NAME --sku basic -g $RG_NAME --location $CLOUD_REGION
+```
+  Example output:
+```
+    [
+    {
+        "adminUserEnabled": false,
+        "anonymousPullEnabled": false,
+        "creationDate": "2021-04-16T21:12:03.724818+00:00",
+        "dataEndpointEnabled": false,
+        "dataEndpointHostNames": [],
+        "encryption": {
+        "keyVaultProperties": null,
+        "status": "disabled"
+        },
+        "id": "/subscriptions/9da62014-f7c1-4e8f-9241-14a2b28362f1/resourceGroups/rg_cswest/providers/Microsoft.ContainerRegistry/registries/csDemoAcr01",
```

Comment:
Do we care if subscription IDs are exposed? Genuinely do not know enough about Azure to make a call on this. Generally, AWS Account IDs are kept secret.

Reply:
Hmm... AWS Account IDs do not change but I could easily kill this subscription and create another in the tenant. Since this is already in the history I will probably just do that and leave it out in future  output. Good catch.